### PR TITLE
network: receive nl80211 multicast messages

### PR DESCRIPTION
**Pull request by yuwata**

### CODE REVIEW

File: src/libsystemd/sd-netlink/netlink-message.c [hunk 1]
```diff
@@ -117,9 +117,10 @@ int sd_netlink_message_request_dump(sd_netlink_message *m, int dump) {
         assert_return(m, -EINVAL);
         assert_return(m->hdr, -EINVAL);
 
-        assert_return(IN_SET(m->hdr->nlmsg_type,
-                             RTM_GETLINK, RTM_GETLINKPROP, RTM_GETADDR, RTM_GETROUTE, RTM_GETNEIGH,
-                             RTM_GETRULE, RTM_GETADDRLABEL, RTM_GETNEXTHOP), -EINVAL);
+        if (m->protocol == NETLINK_ROUTE)
+                assert_return(IN_SET(m->hdr->nlmsg_type,
```
Comment (poettering) [inline src/libsystemd/sd-netlink/netlink-message.c]: doesn't matter, but 
```c
assert_return(m->protocol != NETLINK_ROUTE ||
        IN_SET(…);
```

would make a tiny bit more sense to me

File: src/libsystemd/sd-netlink/sd-netlink.c [hunk 2]
```diff
@@ -445,12 +445,23 @@ static int process_match(sd_netlink *nl, sd_netlink_message *m) {
...
+                for (size_t i = 0; i < c->n_groups; i++)
+                        if (c->groups[i] == m->multicast_group) {
+                                found = true;
+                                break;
+                        }
+
+                if (!found)
+                        continue;
+
+
```
Comment (poettering) [inline src/libsystemd/sd-netlink/sd-netlink.c]: double empty line

File: src/libsystemd/sd-netlink/netlink-genl.c [hunk 3]
```diff
@@ -261,9 +277,10 @@ static int genl_family_get_by_name_internal(
         if (r < 0)
                 return r;
 
-        r = sd_netlink_call(nl, req, 0, &reply);
-        if (r < 0)
-                return r;
+        if (sd_netlink_call(nl, req, 0, &reply) < 0) {
+                (void) genl_family_new_unsupported(nl, name, type_system, ret);
+                return -EOPNOTSUPP;
```
Comment (poettering) [inline src/libsystemd/sd-netlink/netlink-genl.c:280]: So this returns a valid object now on failure? is that necessary? i mean, our coding style dictates taht our functions do not touch return params on failure...

If this really should be kept, then there should be a comment somewhere highlighting this anomaly. But quite frnakly i'd look for a different way to solve this.

i.e. maybe return 0 or 1 depending if the family is supported or not. i.e. return something >= 0 (i.e. sucess) in both cases.

File: src/network/networkd-manager.c [hunk 4]
```diff
@@ -713,15 +706,19 @@ int manager_enumerate(Manager *m) {
                 return log_error_errno(r, "Could not enumerate neighbors: %m");
 
         r = manager_enumerate_nexthop(m);
-        if (r < 0)
-                return log_error_errno(r, "Could not enumerate nexthop rules: %m");
+        if (r == -EOPNOTSUPP || (r == -EINVAL && mac_selinux_enforcing()))
```
Comment (poettering) [inline src/network/networkd-manager.c:766]: hmm what's the brackground of this weird selinux check? maybe could use a comment?

**Review by poettering — COMMENTED** [review]
lgtm, just some minor comments

File: src/network/networkd-manager.c (hunk 4 above)
Comment (yuwata) [reply src/network/networkd-manager.c:766]: It comes from 92b555aaabf710e0a672a7244e8c0e3963075133, Added a brief comment in the code.

File: src/libsystemd/sd-netlink/netlink-genl.c (hunk 3 above)
Comment (yuwata) [reply src/libsystemd/sd-netlink/netlink-genl.c:280]: Ouch. Fixed.

**Comment by yuwata** [comment]
@poettering Thank you for your review. All comments are addressed. Upgrading the green label.

**Pull request merged**